src/test/java/com/wikia/webdriver/testcases/desktop/wampagetests/WamPageTests.java

```diff
@@ -34,25 +38,69 @@ public void wam_002_verifyFilteringByVertical() {
 
     for (WamTab tab : EnumSet.complementOf(EnumSet.of(WamTab.ALL))) {
       wam.selectTab(tab);
+      wam.verifyIfVerticalIdSelectedInUrl(tab.getId());
       wam.verifyWamIndexIsNotEmpty();
       wam.verifyVerticalColumnValuesAreTheSame();
     }
   }
 
-  @RelatedIssue(issueID = "CONCF-6", comment = "Test manually."
-                                               + " Test is failing because WAM page sometimes has wrong order due to duplicate WAM scores.")
+  /**
+   * Test pagination and if WAM ranks are displayed in order
+   */
+  @RelatedIssue(issueID = "DE-4379", comment = "If fails, notify DE team.")
   @Test(groups = {"WamPage003", "WamPageTests", "Smoke5"})
   public void wam_003_verifyPaginationByNextButton() {
-    wam.verifyWamIndexPageFirstColumn(1, 20);
+    wam.verifyWamIndexPageFirstColumnInOrder(1, wam.DEFAULT_WAM_INDEX_ROWS);
     wam.clickNextPaginator();
-    wam.verifyWamIndexPageFirstColumn(21, 40);
+    wam.verifyWamIndexPageFirstColumnInOrder(
+        wam.DEFAULT_WAM_INDEX_ROWS + 1,
+        2 * wam.DEFAULT_WAM_INDEX_ROWS
+    );
+    wam.clickNextPaginator();
+    wam.verifyWamIndexPageFirstColumnInOrder(
+        2 * wam.DEFAULT_WAM_INDEX_ROWS + 1,
+        3 * wam.DEFAULT_WAM_INDEX_ROWS
+    );
+    wam.clickNextPaginator();
+    wam.verifyWamIndexPageFirstColumnInOrder(
+        3 * wam.DEFAULT_WAM_INDEX_ROWS + 1,
+        4 * wam.DEFAULT_WAM_INDEX_ROWS
+    );
   }
 
+  /**
+   * Tests behaviour of date picker
+   */
   @Test(groups = {"wamPage_005", "WamPageTests"})
   public void wam_005_testDatePicker() {
-    wam.verifyLatestDateInDatePicker();
-    String date = "July 12, 2014";
+    wam.verifyDateInDatePicker();
+    // test behaviour of selecting a date
+    String date = "June 1, 2019";
+    wam.typeDateInDatePicker(date);
+    wam.verifyDateInDatePicker(date);
+    wam.verifyWamIndexIsNotEmpty();
+  }
+
+  /**
+   * Tests order of ranks on date that was previously out of order
+   */
+  @Test(groups = {"wamPage_005", "WamPageTests"})
+  public void wam_006_testJune32019DataCorrectness() {
```

Comment:
is it a permanent name for this test?

Reply:
I've added this test since only on this day only I've found an issue that didn't replicate anywhere else.
Recalculating data helped but I think it might be helpful to test it. Do you think it's too specific?

Reply:
I don't know the specifics of your components, so I guess it depends if this is sustainable in terms of testing. For example if this was July 25, 2003, then we would know that it was the Judgment Day, and it would not change, because Judgment Day is inevitable. Maybe it's a Judgment Date or Data - your call ;)

Reply:
In terms of being sustainable, this test should last - data is available for a few years back, e.g. datePicker until now was looking back to 2014. 

If data drops for this specific date in the future we ought to be notified about it.
Since WAM will be sunset/replace in the foreseeable future, we will probably redo tests then anyways.

Thanks for input :)